src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1273,6 +1273,9 @@ abstract class RefChecks extends Transform {
         // The getOrElse part should never happen, it's just here as a backstop.
         reporter.error(pos, sym.compileTimeOnlyMessage getOrElse defaultMsg)
       }
+
+      if (!sym.isMethod && sym.hasAnnotation(VarargsClass))
+        reporter.error(sym.pos, s"${sym.name}: Only methods can be marked @varargs.")
```

Comment:
I know this is a draught-as-in-beer PR, but this warns when you use the undesired method, not when you declare it.

Reply:
Thanks. That saves me from trying it out.

Reply:
I added WIP cream on top.